tests/foreman/cli/test_filter.py

```diff
@@ -0,0 +1,214 @@
+        """
+        org = make_org()
+        # Assign filter to created role
+        filter_ = make_filter({
+            'role-id': self.role['id'],
+            'permissions': self.perms,
+            'organization-ids': org['id'],
+        })
+        for filter_org in filter_['organizations']:
+            self.assertEqual(filter_org, org['name'])
+
+    @tier1
+    def test_positive_create_with_loc(self):
+        """Create a filter and assign it some permissions.
+
+        @id: d7d1969a-cb30-4e97-a9a3-3a4aaf608795
+
+        @Assert: The created filter has the assigned permissions.
+        """
+        loc = make_location()
+        # Assign filter to created role
+        filter_ = make_filter({
+            'role-id': self.role['id'],
+            'permissions': self.perms,
+            'location-ids': loc['id'],
+        })
+        for filter_loc in filter_['locations']:
+            self.assertEqual(filter_loc, loc['name'])
+
+    @tier1
+    def test_positive_delete(self):
+        """Create a filter and delete it afterwards.
+
+        @id: 97d1093c-0d49-454b-86f6-f5be87b32775
+
+        @Assert: The deleted filter cannot be fetched.
+        """
+        filter_ = make_filter({
+            'role-id': self.role['id'],
+            'permissions': self.perms,
+        })
+        Filter.delete({'id': filter_['id']})
+        with self.assertRaises(CLIReturnCodeError):
+            Filter.info({'id': filter_['id']})
+
+    @tier1
+    def test_positive_delete_role(self):
+        """Create a filter and delete the role it points at.
+
+        @id: e2adb6a4-e408-4912-a32d-2bf2c43187d9
+
+        @Assert: The filter cannot be fetched.
+        """
+        filter_ = make_filter({
+            'role-id': self.role['id'],
+            'permissions': self.perms,
+        })
+
+        # A filter depends on a role. Deleting a role implicitly deletes the
+        # filter pointing at it.
+        Role.delete({'id': self.role['id']})
+        with self.assertRaises(CLIReturnCodeError):
+            Role.info({'id': self.role['id']})
+        with self.assertRaises(CLIReturnCodeError):
+            Filter.info({'id': filter_['id']})
+
+    @tier1
+    def test_positive_update_permissions(self):
+        """Create a filter and update its permissions.
+
+        @id: 3d6a52d8-2f8f-4f97-a155-9b52888af16e
+
+        @Assert: Permissions updated.
+        """
+        filter_ = make_filter({
+            'role-id': self.role['id'],
+            'permissions': self.perms,
+        })
+        new_perms = [
+            permission['name']
+            for permission in Filter.available_permissions(
+                {'resource-type': 'User'})
+            ]
+        Filter.update({
+            'id': filter_['id'],
+            'permissions': new_perms
+        })
+        filter_ = Filter.info({'id': filter_['id']})
+        self.assertEqual(filter_['permissions'].split(", "), new_perms)
```

Comment:
please use sets instead of lists as there's no guarantee elements will be returned in exactly the same order